### PR TITLE
Parallelizing Swarm<T>.PreloadAsync()

### PR DESCRIPTION
**Pull request by longfin**
This PR increases the performance of `Swarm<T>.PreloadAsync()` by parallelizing `DialToExistingPeers()`.

### CODE REVIEW

File: Libplanet/Net/Swarm.cs [hunk 1]
```diff
@@ -1062,46 +1060,68 @@ private void BroadcastMessage(Address? except, Message message)
...
+                    else if (!(t.Result is Pong pong))
+                    {
+                        // Mark to skip
+                        return (null, null);
+                    }
+                    else
+                    {
+                        return (peer, pong);
+                    }
+                })
```
Comment (earlbread) [inline Libplanet/Net/Swarm.cs:1111]: How about adding the `cancellationToken` to `ContinueWith`?

File: Libplanet/Net/Swarm.cs [hunk 2]
```diff
@@ -1062,46 +1060,68 @@ private void BroadcastMessage(Address? except, Message message)
...
+
+                if (t.IsFaulted)
                 {
-                    yield return (peer, pong);
+                    throw t.Exception;
                 }
-            }
+
+                return t.Result.Where(pair => !(pair.Item1 is null)).ToArray();
+            });
```
Comment (earlbread) [inline Libplanet/Net/Swarm.cs:1124]: How about adding the `cancellationToken` to `ContinueWith`?

File: Libplanet/Net/Swarm.cs [hunk 3]
```diff
@@ -1062,46 +1060,68 @@ private void BroadcastMessage(Address? except, Message message)
...
+                    }
+                    else if (t.IsCanceled)
+                    {
+                        // Mark to skip
+                        return (null, null);
+                    }
+                    else if (!(t.Result is Pong pong))
+                    {
+                        // Mark to skip
+                        return (null, null);
```
Comment (earlbread) [inline Libplanet/Net/Swarm.cs:1105]: How about combining these multiple conditions into one?

**Comment by longfin** [comment]
Amended a commit to applying the suggestion (https://github.com/planetarium/libplanet/pull/846/commits/7d7bf047c15e48c095a0225ead6a612037b03322)

**Pull request merged**